### PR TITLE
Updating the page that describes Scripts

### PR DESCRIPTION
**Pull request by vishalchangrani**
This is in preparation for the launch of Script execution on AN.

Current page: https://developers.flow.com/build/basics/scripts
Updated page: https://docs-git-vishal-scripts-onflow.vercel.app/build/basics/scripts

### CODE REVIEW

File: docs/build/basics/scripts.md [hunk 1]
```diff
@@ -4,24 +4,40 @@ sidebar_position: 4
...
-Scripts can return a typed value:
+Scripts can return a typed value.
 
 ```cadence
 pub fun main(): Int {
 	return 1 + 2
 }
 ```
 
+Scripts can also accept arguments.
```
Comment (SeanRobb) [inline docs/build/basics/scripts.md:33]: replace `.` with `:` to align with other segments

File: docs/build/basics/scripts.md [hunk 2]
```diff
@@ -4,24 +4,40 @@ sidebar_position: 4
...
 ```cadence
-pub fun main() {}
+// The 'main' function is the entry point function and every script needs to have one.
+pub fun main() {
+ // Cadence statements to be executed go here
+}
 ```
 
-Scripts can return a typed value:
+Scripts can return a typed value.
```
Comment (SeanRobb) [inline docs/build/basics/scripts.md:25]: replace `.` with `:` to align with other segments

**Review by SeanRobb — CHANGES_REQUESTED** [review]

File: docs/build/basics/scripts.md [hunk 3]
```diff
@@ -4,14 +4,22 @@ sidebar_position: 4
 
 # Scripts
 
-A script is executable Cadence code that queries the Flow network but does not modify it. Unlike Flow transactions, they don’t need signing and they can return a value. You can think of executing scripts as a read-only operation. 
+A script provides a light-weight method to query chain data.
 
-Scripts are executed on Access Nodes or Execution Nodes. 
+It is an executable Cadence code that can query for Flow execution state data but cannot modify it in any way.
```
Comment (peterargue) [inline docs/build/basics/scripts.md:9]:
```suggestion
It is executable Cadence code that can query for Flow execution state data but cannot modify it in any way.
```

File: docs/build/basics/scripts.md [hunk 4]
```diff
@@ -57,3 +76,27 @@ There are multiple SDKs implementing the above APIs for different languages:
...
+
+```sh
+flow scripts execute ./helloWorld.cdc
+```
+
+A user can define their own scripts or can use already defined scripts by the contract authors that can be found by using the [FLIX](../../tools/flow-cli/flix) service.
+
+## Limitations
+
+1. **Rate limit** - Script execution is subjected to API rate-limits imposed by the Access nodes and the Execution nodes. The rate limits for the Public Access nodes hosted by QuickNode are mentioned [here](https://www.quicknode.com/docs/flow#endpoint-rate-limits).
```
Comment (peterargue) [inline docs/build/basics/scripts.md:92]:
```suggestion
1. **Rate limit** - Script execution is subjected to API rate-limits imposed by the Access nodes and the Execution nodes. The rate limits for the Public Access nodes hosted by QuickNode are outlined [here](https://www.quicknode.com/docs/flow#endpoint-rate-limits).
```

File: docs/build/basics/scripts.md [hunk 5]
```diff
@@ -57,3 +76,27 @@ There are multiple SDKs implementing the above APIs for different languages:
...
+
+
+2. **Computation limit** - Similar to a transaction, each script is also subjected to a computation limit. Currently, the compute (gas) limit for a script is 100,000.
```
Comment (peterargue) [inline docs/build/basics/scripts.md:95]:
```suggestion
2. **Computation limit** - Similar to a transaction, each script is also subjected to a computation limit. The specific value can be configured by individual Access and Execution node operators. Currently, the default compute (gas) limit for a script is 100,000.
```

File: docs/build/basics/scripts.md [hunk 6]
```diff
@@ -57,3 +76,27 @@ There are multiple SDKs implementing the above APIs for different languages:
...
+
+
+3. **Historic block data limit**
+   1. Script execution on the execution node is restricted to approximately the last 600 blocks. Any request for script execution on an execution node on a past block (specified by block ID or block height) will fail if that block is more than 600 blocks in the past.
```
Comment (peterargue) [inline docs/build/basics/scripts.md:99]: it's actually ~100 blocks these days
```suggestion
   1. Script execution on execution nodes is restricted to approximately the last 100 blocks. Any request for script execution on an execution node on a past block (specified by block ID or block height) will fail if that block is more than 100 blocks in the past.
```

File: docs/build/basics/scripts.md [hunk 7]
```diff
@@ -57,3 +76,27 @@ There are multiple SDKs implementing the above APIs for different languages:
...
+   2. Script execution on an access node can go much beyond the last 600 blocks but is restricted to the height when the [last](https://developers.flow.com/networks/node-ops/node-operation/past-sporks) network upgrade ([HCU](https://developers.flow.com/networks/node-ops/node-operation/hcu) or spork) occurred.
```
Comment (peterargue) [inline docs/build/basics/scripts.md:100]:
```suggestion
   2. Script execution on an access node can go much beyond the last 100 blocks but is restricted to the height when the [last](https://developers.flow.com/networks/node-ops/node-operation/past-sporks) network upgrade ([HCU](https://developers.flow.com/networks/node-ops/node-operation/hcu) or spork) occurred. 
```

**Review by peterargue — APPROVED** [review]

**Pull request merged**